packages/ui-overlays/src/Overlay/props.ts

```diff
@@ -28,153 +28,215 @@ import PropTypes from 'prop-types'
 import { element } from '@instructure/ui-prop-types'
 import { transitionTypePropType } from '@instructure/ui-motion'
 
-import type { PortalNode } from '@instructure/ui-portal'
+import type { PortalNode, PortalProps } from '@instructure/ui-portal'
 import type { PositionMountNode } from '@instructure/ui-position'
-import type { TransitionType } from '@instructure/ui-motion'
+import type { TransitionProps, TransitionType } from '@instructure/ui-motion'
+import type { DialogProps } from '@instructure/ui-dialog'
 import type {
   OtherHTMLAttributes,
-  PropValidators
+  PickPropsWithExceptions,
+  PropValidators,
+  UIElement
 } from '@instructure/shared-types'
 
 type OverlayOwnProps = {
+  /**
+   * Whether or not the `<Overlay />` is open
+   */
   open?: boolean
-  onOpen?: (DOMNode: PortalNode) => any
-  onClose?: (...args: any[]) => any
-  mountNode?: PositionMountNode
-  insertAt?: 'bottom' | 'top'
-  label: string
-  onDismiss?: (...args: any[]) => any
-  defaultFocusElement?: React.ReactElement | ((...args: any[]) => any)
-  applicationElement?:
-    | React.ReactElement[]
-    | React.ReactElement
-    | ((...args: any[]) => any)
-  contentElement?: React.ReactElement | ((...args: any[]) => any)
-  shouldContainFocus?: boolean
-  shouldReturnFocus?: boolean
-  shouldCloseOnDocumentClick?: boolean
-  shouldCloseOnEscape?: boolean
-  transition?: TransitionType
-  in?: boolean
-  unmountOnExit?: boolean
-  transitionOnMount?: boolean
-  transitionEnter?: boolean
-  transitionExit?: boolean
-  onEnter?: (...args: any[]) => any
-  onEntering?: (...args: any[]) => any
-  onEntered?: (...args: any[]) => any
-  onExit?: (...args: any[]) => any
-  onExiting?: (...args: any[]) => any
-  onExited?: (type?: TransitionType) => void
-  children?: React.ReactNode
-}
-
-type PropKeys = keyof OverlayOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
 
-type OverlayProps = OverlayOwnProps & OtherHTMLAttributes<OverlayOwnProps>
+  /**
+   * The type of `<Transition />` to use for animating in/out
+   */
+  transition?: TransitionType // passed as to Transition as `type`
 
-const propTypes: PropValidators<PropKeys> = {
-  children: PropTypes.node,
+  // TODO: deprecate applicationElement, it was removed from Dialog in v6
```

Comment:
Can you please make a ticket of this one? Or just fix here, as I see setting this value does nothing

Reply:
Deprecated it.